lib/iris/tests/unit/coords/test_Coord.py

```diff
@@ -355,6 +356,112 @@ def test_different_array_attrs_incompatible(self):
+        self.bounds = np.array([[0, 2], [2, 4]])
+
+    def test_no_bounds(self):
+        coord = iris.coords.DimCoord(self.points)
+        result = coord.is_contiguous()
+        self.assertFalse(result)
+
+    def test_1d_coord_contiguous(self):
+        coord = iris.coords.DimCoord(self.points, bounds=self.bounds)
+        result = coord.is_contiguous()
+        self.assertTrue(result)
+
+    def test_1d_coord_discontiguous(self):
+        coord = iris.coords.DimCoord(self.points, bounds=[[0, 2], [3, 4]])
+        result = coord.is_contiguous()
+        self.assertFalse(result)
+
+    def test_1d_coord_contiguous_atol(self):
+        coord = iris.coords.DimCoord(self.points/1000, bounds=self.bounds/1000)
+        result = coord.is_contiguous(atol=1e-2)
+        self.assertTrue(result)
+
+
+class Test__sanity_check_bounds(tests.IrisTest):
+    def test_coord_1d_2_bounds(self):
+        coord = iris.coords.DimCoord([0, 1], standard_name='latitude',
+                                     bounds=np.array([[0, 1], [1, 2]]))
+        coord._sanity_check_bounds()
+
+    def test_coord_1d_no_bounds(self):
+        coord = iris.coords.DimCoord([0, 1], standard_name='latitude')
+        emsg = "Invalid operation for 'latitude', with 0 bounds. Contiguous " \
+               "bounds are only defined for 1D coordinates with 2 bounds."
+        with self.assertRaisesRegexp(ValueError, emsg):
+            coord._sanity_check_bounds()
+
+    def test_coord_1d_1_bounds(self):
+        coord = iris.coords.DimCoord([0, 1], standard_name='latitude',
+                                     bounds=np.array([[0], [1]]))
+        emsg = "Invalid operation for 'latitude', with 1 bounds. Contiguous " \
+               "bounds are only defined for 1D coordinates with 2 bounds."
+        with self.assertRaisesRegexp(ValueError, emsg):
+            coord._sanity_check_bounds()
+
+    def test_coord_2d_4_bounds(self):
+        coord = iris.coords.AuxCoord(
+            [[0, 0], [1, 1]], standard_name='latitude',
+            bounds=np.array([[[0, 0, 1, 1], [0, 0, 1, 1]],
+                             [[1, 1, 2, 2], [1, 1, 2, 2]]]))
+        coord._sanity_check_bounds()
+
+    def test_coord_2d_no_bounds(self):
+        coord = iris.coords.AuxCoord([[0, 0], [1, 1]],
+                                     standard_name='latitude')
+        emsg = "Invalid operation for 'latitude', with 0 bounds. Contiguous " \
+               "bounds are only defined for 2D coordinates with 4 bounds."
+        with self.assertRaisesRegexp(ValueError, emsg):
+            coord._sanity_check_bounds()
+
+    def test_coord_2d_2_bounds(self):
+        coord = iris.coords.AuxCoord([0, 1], standard_name='latitude',
+                                     bounds=np.array([[0], [1]]))
+        emsg = "Invalid operation for 'latitude', with 1 bounds. Contiguous " \
+               "bounds are only defined for 1D coordinates with 2 bounds."
+        with self.assertRaisesRegexp(ValueError, emsg):
```

Comment:
Shouldn't this raise an error about 2d coordinates needing 4 bounds, rather than 1d coordinates needing 2 bounds?